### DISCUSSION TITLE
An unsupported feature is required for this Sigma rule

### DISCUSSION THREAD

**Discussion by presianbg** [Security-Onion-Solutions/securityonion#3538] [Unsupported Versions] [answered]
Hi,

SOC Prime and Corelight created and shared some free SIGMA rules on the [Threat Detection Marketplace](https://tdm.socprime.com), but those rules can't be used in the playbook, because:

```
An unsupported feature is required for this Sigma rule (/tmp/tmpuvx5omn5): Aggregations not implemented for this backend
Feel free to contribute for fun and fame, this is open source :) -> https://github.com/Neo23x0/sigma
```
The condition of the rule is:
`condition: selection AND NOT exclude_history_payload | count(id_resp_p) by id_orig_h > 25`

and if we remove the `count(id_resp_p) by id_orig_h > 25`  the rule is successfully converted.

 Looks like the grouping was broken, but got fixed:
 https://github.com/SigmaHQ/sigma/issues/940

I can't figure out if such rules can be used in SecurityOnion's Playbook. 

Any thoughts on this ?

Cheers,
Presian

**Comment by presianbg** [comment, marked as answer]
Also I'm not able to make this rule work:
https://github.com/SigmaHQ/sigma/blob/master/rules/network/zeek/zeek_rdp_public_listener.yml

If imported (by hand) and activated, its starting to match every connection:
![image](https://user-images.githubusercontent.com/2163486/111685408-8e2c2e80-8830-11eb-89a5-902166ea7753.png)

![image](https://user-images.githubusercontent.com/2163486/111687386-e5330300-8832-11eb-8ad1-9de2d44eebe6.png)


Looks like `service: rdp` is not taken under consideration.

EDIT: For some reason resulting Elasticsearch Query is not working too:
![image](https://user-images.githubusercontent.com/2163486/111686730-14954000-8832-11eb-9add-ad4ab3c73d75.png)


![image](https://user-images.githubusercontent.com/2163486/111687344-d8aeaa80-8832-11eb-8a86-999a9023c2dd.png)

**Reply by defensivedepth** [reply]
I need to dig into the Agg stuff a bit - not sure what is up with that.

The RDP/IP field mapping issues will be fixed in our next release (2.2.40).

**Reply by presianbg** [reply]
Just to confirm that the RDP playbook works as expected now :100:

**Reply by defensivedepth** [reply]
@presianbg Great! Let us know if you see other issues like that.

**Reply by presianbg** [reply]
Sure, are you going to  investigate the  `An unsupported feature is required for this Sigma rule (/tmp/tmpuvx5omn5): Aggregations not implemented for this backend` part ?

**Reply by defensivedepth** [reply]
@presianbg here is the deal:

When you click on Convert in the `Create New Play` window in Playbook, it converts it into an Elasticsearch query that you can copy and paste directly into Hunt / Kibana to test.

When you click "Create Play" and it creates the Play, Playbook converts it into an Elastalert rule.

The Elasticsearch sigma backend does not support Aggs, but Elastalert does.
